src/jaegertracing/Tracer.h

```diff
@@ -201,12 +201,20 @@ class Tracer : public opentracing::Tracer,
     }
 
   private:
+    using TextMapPropagator =
+        propagation::Propagator<const opentracing::TextMapReader&,
+                                const opentracing::TextMapWriter&>;
+    using HTTPHeaderPropagator =
+        propagation::Propagator<const opentracing::HTTPHeadersReader&,
+                                const opentracing::HTTPHeadersWriter&>;
+
     Tracer(const std::string& serviceName,
            const std::shared_ptr<samplers::Sampler>& sampler,
            const std::shared_ptr<reporters::Reporter>& reporter,
            const std::shared_ptr<logging::Logger>& logger,
            const std::shared_ptr<metrics::Metrics>& metrics,
-           const propagation::HeadersConfig& headersConfig,
+           const TextMapPropagator* textPropagator,
```

Comment:
my C++ is quite rusty, but why is it that the other composable objects above are captured via shared_ptr and propagators via raw pointer?

Reply:
Mine is rusty too, but I changed it to shared_ptr.